### ISSUE TITLE
Task 3.1: Create 20% Sampled Dataset for Model Development

### ISSUE THREAD

**Issue by rishabhkaushal07** [sfu-bigdata/Claude-demo2-Vancouver-Business-Heritage#7]
## Objective
Create a smaller working dataset by randomly sampling 20% of the full processed data. This enables faster iteration during model development while maintaining class distribution.

**Part of #4** (Task 3: Model Development)

## Checklist
- [ ] Load full processed dataset from `data/processed/businesses_with_heritage_labels.csv`
- [ ] Use stratified sampling to maintain class distribution (98.9%/1.1%)
- [ ] Set fixed random seed (`random_state=42`) for reproducibility
- [ ] Sample 20% of data (~11,253 records)
- [ ] Save sampled dataset to `data/processed/businesses_sampled_20pct.csv`
- [ ] Document sampling parameters in script

## Technical Notes
- Use `sklearn.model_selection.train_test_split` with `stratify` parameter
- Full dataset: 56,267 records → Sampled: ~11,253 records
- Class distribution must be preserved: 98.9% near heritage (1), 1.1% not near (0)

## Acceptance Criteria
- Sampled dataset saved to `data/processed/businesses_sampled_20pct.csv`
- Class distribution preserved (verified with value_counts)
- Sampling is reproducible with same seed
- Script saved as `scripts/create_sample.py`

## Dependencies
- **Blocks**: Tasks 3.2, 3.3, 3.4 (model training sub-issues)

**Comment by rishabhkaushal07** [comment]
Task 3.1: Create 20% Sampled Dataset completed successfully.

## Completed Deliverables
- ✅ Loaded full processed dataset (56,267 records)
- ✅ Used stratified sampling to maintain class distribution
- ✅ Set fixed random seed (`random_state=42`) for reproducibility
- ✅ Sampled 20% of data (11,254 records)
- ✅ Saved sampled dataset to `data/processed/businesses_sampled_20pct.csv`
- ✅ Created sampling script: [`scripts/create_sample.py`](https://github.com/sfu-bigdata/Claude-demo2-Vancouver-Business-Heritage/blob/main/scripts/create_sample.py)

## Results
- **Sample size**: 11,254 records (2.77 MB)
- **Class distribution preserved perfectly**:
  - Not near heritage (0): 120 records (1.07%)
  - Near heritage (1): 11,134 records (98.93%)
  - Max distribution difference: 0.00%

## Verification
```
Original: 601 (1.07%) / 55,666 (98.93%)
Sampled:  120 (1.07%) / 11,134 (98.93%)
```

## Next Steps
The sampled dataset is now ready for model training. Sub-issues #8 (Logistic Regression), #9 (Random Forest), and #10 (XGBoost) can proceed **in parallel**.
